src/daria/analysis/segmentationcomparison.py

```diff
@@ -1,113 +1,471 @@
+
+    def __call__(
+        self,
+        *segmentations,
+        plot_result: bool = False,
+        **kwargs,
+    ) -> np.ndarray:
+        """
+        Comparison of segmentations.
+
+        Args:
+            segmentations (asterisk argument): Allows to provide an arbitraty
+                number of segmented numpy arrays or da.Images of integers to be compared
+            Optional keyword arguments (kwargs):
+                plot_result (bool): plots the result with matplotlib if True,
+                    default is False.
+                roi (tuple): roi where the segmentations should be compared,
+                    if default the maximal roi that fits in all segmentations
+                    is chosen.
+
+        """
+
+        # Define number of segmentations
+        assert self.number_of_segmented_images == len(segmentations)
+
+        if "roi" in kwargs:
+            roi = kwargs["roi"]
+            return_image: np.ndarray = np.zeros(
+                (roi[0].stop - roi[0].start, roi[1].stop - roi[1].start) + (3,),
+                dtype=np.uint8,
+            )
+
+        else:
+            if isinstance(segmentations[0], np.ndarray):
+                nx = min([seg.shape[0] for seg in segmentations])
+                ny = min([seg.shape[1] for seg in segmentations])
+            elif isinstance(segmentations[0], da.Image):
+                nx = min([seg.img.shape[0] for seg in segmentations])
+                ny = min([seg.img.shape[1] for seg in segmentations])
+            roi = (slice(0, nx), slice(0, ny))
+            return_image: np.ndarray = np.zeros((nx, ny) + (3,), dtype=np.uint8)
+
+        # Enter gray everywhere there are ovelaps of different segmentations
+
+        if isinstance(segmentations[0], np.ndarray):
+            for k in range(self.number_of_segmented_images):
+                for i in range(k + 1, self.number_of_segmented_images):
+
+                    # Overlap of components
+                    for c_num, c in enumerate(self.components):
+                        return_image[
+                            np.logical_and(
+                                segmentations[k][roi] == c,
+                                segmentations[i][roi] == c,
+                            )
+                        ] += self.gray_base[c_num]
+
+                    # Overlap of different components
+                    return_image[
+                        np.logical_or(
+                            np.logical_and(
+                                segmentations[k][roi] == self.components[0],
+                                segmentations[i][roi] == self.components[1],
+                            ),
+                            np.logical_and(
+                                segmentations[k][roi] == self.components[1],
+                                segmentations[i][roi] == self.components[0],
+                            ),
+                        )
+                    ] += self.gray_base[2]
```

Comment:
To be consistent with the last assignment
```suggestion
                    ] = self.gray_base[2]
```

Reply:
I am not sure what you mean?

Reply:
I did not get why you use summation here? after all, it seemed you treat disjoint cases anyhow. Overwriting seemed most natural. If I got something wrong, simply ignore my comment.

Reply:
My "secret" plan here, is that with the summation I can check how many are overlapping, and not just whether some are overlapping or not. I am almost sure that it will be too much information in the end (with 5 compared images), but I will keep it for now, and probably revert to the equality at a later point.

Reply:
I tested with 4 images now, and it became a true mess. I will make the equality instead.

Reply:
Ah OK. In principle I am fine with these type of hacks. But then, good documentation is needed - for the future hacks.